ddtrace/profiling/_threading.pyx

```diff
@@ -1,52 +1,64 @@
 from __future__ import absolute_import
 
-import threading
+import sys
+import threading as ddtrace_threading
 import typing
 import weakref
 
 import attr
-
-from ddtrace.internal import nogevent
+from six.moves import _thread
 
 
 cpdef get_thread_name(thread_id):
-    # This is a special case for gevent:
-    # When monkey patching, gevent replaces all active threads by their greenlet equivalent.
-    # This means there's no chance to find the MainThread in the list of _active threads.
-    # Therefore we special case the MainThread that way.
-    # If native threads are started using gevent.threading, they will be inserted in threading._active
-    # so we will find them normally.
-    if thread_id == nogevent.main_thread_id:
-        return "MainThread"
-
-    # We don't want to bother to lock anything here, especially with eventlet involved 😓. We make a best effort to
-    # get the thread name; if we fail, it'll just be an anonymous thread because it's either starting or dying.
-    try:
-        return threading._active[thread_id].name
-    except KeyError:
+    # Do not force-load the threading module if it's not already loaded
```

Comment:
In what scenarios will `threading` not be found in sys.modules? Do we expect this to happen frequently?

Reply:
Since we are unloading modules when gevent is detected, we don't know if and when the threading module will be loaded on every call to this function. We really want to ensure we use the `threading` module that the application uses but without force-loading, so our best bet is to look into `sys.modules`.